src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -504,29 +519,63 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetReachable(bool & rea
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetUniqueId(char * buf, size_t bufSize)
 {
-    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    CHIP_ERROR err;
+    size_t uniqueIdLen = 0; // without counting null-terminator
+    err                = ReadConfigValueStr(ConfigClass::kConfigKey_UniqueId, buf, bufSize, uniqueIdLen);
+
+    ReturnErrorOnFailure(err);
+
+    ReturnErrorCodeIf(uniqueIdLen >= bufSize, CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(buf[uniqueIdLen] != 0, CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    return err;
 }
 
 template <class ConfigClass>
-CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetLifetimeCounter(uint16_t & lifetimeCounter)
+CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::StoreUniqueId(const char * uniqueId, size_t uniqueIdLen)
 {
+    return WriteConfigValueStr(ConfigClass::kConfigKey_UniqueId, uniqueId, uniqueIdLen);
+}
+
+template <class ConfigClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GenerateUniqueId(char * buf, size_t bufSize)
+{
+    uint64_t randomUniqueId = Crypto::GetRandU64();
+    return Encoding::BytesToUppercaseHexString(reinterpret_cast<uint8_t *>(&randomUniqueId), sizeof(uint64_t), buf, bufSize);
+}
+
 #if CHIP_ENABLE_ROTATING_DEVICE_ID
+template <class ConfigClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetLifetimeCounter(uint16_t & lifetimeCounter)
+{
     lifetimeCounter = static_cast<uint16_t>(mLifetimePersistedCounter.GetValue());
     return CHIP_NO_ERROR;
-#else
-    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
-#endif
 }
 
 template <class ConfigClass>
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::IncrementLifetimeCounter()
 {
-#if CHIP_ENABLE_ROTATING_DEVICE_ID
     return mLifetimePersistedCounter.Advance();
+}
+
+template <class ConfigClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetRotatingDeviceIdUniqueId(MutableByteSpan & uniqueIdSpan)
+{
+#ifdef CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID
```

Comment:
could we place the entire method into an IFDEF so it does not even compile if this is not defined instead of returning NOT_IMPLEMENTED?

Less code and the virtual methods are already behind a ifdef

Reply:
This method is already in the same IFDEF, as virtual methods: https://github.com/project-chip/connectedhomeip/pull/15031/files#diff-5877c3174710d184cd07eb3e96066275f2934b721b547519e59a7ba06a796af5R547

The CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID is just about providing the value of an ID, not enabling rotating device ID feature.

Reply:
CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID should not be available if ENABLE_ROTATING_DEVICE_ID is not enabled right?

The reason I am sensitive here: rotating device id is an optional spec feature and I want to make sure it stays optional in the SDK. We historically failed to ensure that and had a harder time fixing because compilation without rotating device ID was failing.

As a result, I expect any rotation device id code to be behind an ifdef. This includes a `GetRotating*` calls.

Reply:
It should be done now